### PR TITLE
wtclient: automatically initialise while creating a watchtower server

### LINKED ISSUE #5344: Activate watchtower client by default
### Background

I think it would be useful if the watchtower client is enabled by default. More people should use watchtowers I feel; but them having to activate wtclient first makes it harder. Why not enable by default?

### Your environment

* version of `lnd` commit=v0.12.1-beta

### Steps to reproduce

People try to connect to watchtower
People get error `watchtower not active`

### Expected behaviour

People try to connect to watchtower
Watchtower connects

### Actual behaviour

Well I just said that ;).

### PR DESCRIPTION
**Pull request by DarthBenro008**
- Fixes #5344 

Initially, the `--wtclient.active` had to be passed alongside `--watchtower.active` for `lncli` to be able to connect to the watchtower

This PR inverts the functionality, now `wtclient` is automatically created, however if you do not want it it to create automatically, you can pass `--watchtower.deactivate` now instead.

**Old Command:**
`lnd --watchtower.active --wtclient.activate`

**New Command (automatically creates a wtclient):** 
`lnd --watchtower.active`

**New Command to disable creating wtclient:**
`lnd --watchtower.active --wtclient.deactivate`

**[Test]:** personally tested on local machine and passes `go test .../..`

#### Pull Request Checklist

- [X] All changes are Go version 1.15 compliant
- [ ] Your PR passes all CI checks. If a check cannot be passed for a justifiable reason, that reason must be stated in the commit message and PR description.
- [X] If this is your first time contributing, we recommend you read the [Code Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md)
   - [X] The code being submitted is commented according to [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation)
   - [X] Commits have a logical structure according to [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure)
- [ ] For new code: Code is accompanied by tests which exercise both the positive and negative (error paths) conditions (if applicable)
- [ ] For bug fixes: If possible, code is accompanied by new tests which trigger the bug being fixed to prevent regressions
- [ ] Any new logging statements use an appropriate subsystem and logging level
- [X] For code and documentation: lines are wrapped at 80 characters (the tab character should be counted as 8 characters, not 4, as some IDEs do per default)

### CODE REVIEW

**Review by carlaKC — COMMENTED** [review]
Thanks for the PR! 

Ideally we find a way to address this without a breaking change to config. 
What do you think about leaving the `active` flag as is + defaulting to true? We can then add a `deactivate`, as done in this PR, for people who'd like to turn it off. 

Also noticed that the CI "every commit builds" and "sample config" checks failed on this one (likely need to add the `deactivate` flag to our sample config).

**Comment by carlaKC** [comment]
> Can we add a deprecation notice and warn the user about that they do not need to pass it anymore and eventually clean it up in future versions 

Yeah, was thinking exactly this :)

**Comment by DarthBenro008** [comment]
Hi @carlaKC , sorry for the delay, apparently the codebase that handles the database for watchtower and wtclient had completely changed, hence the merge conflicts. I re-wrote the implementation and have force pushed the codebase. 

- I have left `--wtclient.activate` as it is and deprecated it
- I have tested all the changes locally

File: lncfg/wtclient.go [hunk 1]
```diff
@@ -15,6 +15,11 @@ type WtClient struct {
 	// SweepFeeRate specifies the fee rate in sat/byte to be used when
 	// constructing justice transactions sent to the tower.
 	SweepFeeRate uint64 `long:"sweep-fee-rate" description:"Specifies the fee rate in sat/byte to be used when constructing justice transactions sent to the watchtower."`
+
+	// Active determines whether a watchtower client should be created to
+	// back up channel states with registered watchtowers.
+	// Deprecated. Kept for migration purposes. Use Deactivate instead
+	Active bool `long:"active" description:"Whether the daemon should use private watchtowers to back up revoked channel states."`
```
Comment (carlaKC) [inline lncfg/wtclient.go:22]: Set a default value for this to true?

**Review by carlaKC — COMMENTED** [review]
Forgot to click submit on my review 🙈 

Just some nits, will test this out in the meantime.

**Comment by carlaKC** [comment]
Closing due to inactivity

**Pull request closed without merging**